### PR TITLE
[4.5] docs(dropdownbutton): Documentation for DropDownButton

### PR DESCRIPTION
**Pull request by ntacheva**

### CODE REVIEW

File: components/dropdownbutton/icons.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+---
+title: Icons
+page_title: DropDownButton Icons
+description: How to use icons with the DropDownButton for Blazor.
```
Comment (yordan-mitev) [inline components/dropdownbutton/icons.md:4]: See the comment in `overview.md`.

File: components/dropdownbutton/icons.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+slug: dropdownbutton-icons
+tags: telerik,blazor,dropdownbutton,icons
+published: True
+position: 5
+---
+
+# DropDownButton Icons
+
+The `<TelerikDropDownButton>` and `<DropDownButtonItem>` provide an `Icon` parameter to add an icon to the main button and all secondary action items.
```
Comment (yordan-mitev) [inline components/dropdownbutton/icons.md:13]:
```suggestion
The `<TelerikDropDownButton>` and `<DropDownButtonItem>` provide an `Icon` parameter that allows you to add an icon to the main button and all secondary action items.
```

File: components/dropdownbutton/icons.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        background: purple;
+    }
+</style>
+
+@code {
+    string CustomIconClass { get; set; } = "my-icon";
+}
+````
+
+>tip You can also add custom icons and images with additional markup inside the `<DropDownButtonContent>`, where the text is.
```
Comment (yordan-mitev) [inline components/dropdownbutton/icons.md:50]: The statement is somewhat ambiguous - "with" in "custom icons and images with additional markup" can also mean "through". If this is the case, you must phrase it.

File: components/dropdownbutton/icons.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+>note Images used as icons should generally be small enough to fit in a line of text. The primary button is an inline element and is not designed for large images. If you want to use big icon buttons, consider one of the following options:
```
Comment (yordan-mitev) [inline components/dropdownbutton/icons.md:52]: This is a huge note that looks funny once you build the article. Consider turning it into a new document section by adding an H2, for example, "## Best Practices".

image

File: components/dropdownbutton/icons.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+>
+> * Define a `Class` on the button that provides `height` and `width`. The width and height can be set in `px` sufficient to accommodate the icon or to `auto`,
```
Comment (yordan-mitev) [inline components/dropdownbutton/icons.md:54]:
```suggestion
> * Define a `Class` for the button that provides `height` and `width`. The width and height can be set to `auto` or to a value in `px`. When using a static value, it must be sufficient to accommodate the icon.
```

File: components/dropdownbutton/appearance.md [hunk 6]
```diff
@@ -0,0 +1,240 @@
+---
+title: Appearance
+page_title: DropDownButton Appearance
+description: Appearance settings of the DropDownButton for Blazor.
```
Comment (yordan-mitev) [inline components/dropdownbutton/appearance.md:4]: See the comment in `overview.md`.

File: components/dropdownbutton/icons.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
+---
+title: Icons
+page_title: DropDownButton Icons
```
Comment (yordan-mitev) [inline components/dropdownbutton/icons.md:3]: This seems like a pattern used by other feature-related articles.
```suggestion
page_title: Blazor DropDownButton - Icons
```

File: components/dropdownbutton/appearance.md [hunk 8]
```diff
@@ -0,0 +1,240 @@
+---
+title: Appearance
+page_title: DropDownButton Appearance
```
Comment (yordan-mitev) [inline components/dropdownbutton/appearance.md:3]:
```suggestion
page_title: Blazor DropDownButton - Appearance
```

File: components/dropdownbutton/appearance.md [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+tags: telerik,blazor,dropdownbutton,appearance,styling
+published: True
+position: 10
+---
+
+# DropDownButton Appearance
+
+This article describes the declarative settings of the DropDownButton component, which affect the styling and appearance of the primary button.
+
+The DropDownButton provides the same appearance parameters, as the regular [Button component]({%slug button-appearance%}):
```
Comment (yordan-mitev) [inline components/dropdownbutton/appearance.md:15]:
```suggestion
The DropDownButton provides the same appearance parameters as the regular [Button component]({%slug button-appearance%}):
```

File: components/dropdownbutton/appearance.md [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+
+* [FillMode](#fillmode)
+* [Rounded](#rounded)
+* [Size](#size)
+* [ThemeColor](#themecolor)
+
+
+## Setting Parameter Values
+
+The examples in this article use **reflection** to show all possible values of the DropDownButton parameters. In a real-world scenario, you can use two options to set the desired parameter values:
```
Comment (yordan-mitev) [inline components/dropdownbutton/appearance.md:25]: Why bold?
```suggestion
The examples in this article use reflection to show all possible values of the DropDownButton parameters. In a real-world scenario, you can use two options to set the desired parameter values:
```

File: components/dropdownbutton/appearance.md [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+            .GetFields().ToList();
+
+        base.OnInitialized();
+    }
+}
+````
+
+## Size
+
+The `Size` parameter can change some dimensions of the DropDownButton's primary button, such as height, margins or paddings. Possible values are the `string` members of the static class `ThemeConstants.DropDownButton.Size`.
```
Comment (yordan-mitev) [inline components/dropdownbutton/appearance.md:142]:
```suggestion
The `Size` parameter can change some dimensions of the DropDownButton's primary button, such as height, margins, or paddings. Possible values are the `string` members of the static class `ThemeConstants.DropDownButton.Size`.
```

File: components/dropdownbutton/events.md [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+description: How to handle events with the DropDownButton for Blazor.
+slug: dropdownbutton-events
+tags: telerik,blazor,dropdownbutton,events
+published: True
+position: 15
+---
+
+# DropDownButton Events
+
+This article describes the DropDownButton events:
```
Comment (yordan-mitev) [inline components/dropdownbutton/events.md:13]: - A list with a single item looks unnatural and wastes space.
- An additional heading isn't required for such a short article. Such structure makes sense when the component exposes two or more events.
```suggestion
The DropDownButton exposes an [`OnClick`](#onclick) event that you can use to initiate an action within the application.
```

File: components/dropdownbutton/events.md [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+
+* [OnClick](#onclick)
```
Comment (yordan-mitev) [inline components/dropdownbutton/events.md:15]:
```suggestion
```

File: components/dropdownbutton/events.md [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+
+## OnClick
```
Comment (yordan-mitev) [inline components/dropdownbutton/events.md:17]:
```suggestion
```

**Review by dimodi — CHANGES_REQUESTED** [review]
- [ ] Add the component to the home page list.
- [ ] The overview description is identical to the one of SplitButton. Explain the differences and cross-link the components at both places.
- [x] Add an `OnClick` handler to the component itself in the Basic example
- [x] Use private accessor everywhere where it is missing
- [ ] (Icons) "The SplitButton provides four parameters and five different ways..." - when you start like this, this implies a list of options. Are the parameters really 4? Which are they?
- [x] `("sln")` - the reader won't understand what is this icon and when it should be used. Normally, we shouldn't advertise this anymore.
- [ ] (Events) Does `OnClick` provide event arguments?

File: components/dropdownbutton/icons.md (hunk 7 above)
Comment (ntacheva) [reply components/dropdownbutton/icons.md:3]: That's correct, there is such a patter. However, we do not include the Blazor keyword in the page title. I will update the titles of all articles.

**Review by dimodi — APPROVED** [review]

**Pull request merged**